### PR TITLE
feat(kafka:status): RHICOMPL-1940 use kafka.topics for connectivity

### PR DESCRIPTION
**Pull request by akofink**
The kafka status check used to send a test message. Instead, we can just
ask the kafka instance for the available topics, which makes some kind
of metadata request that will fail if kafka is not available.

Signed-off-by: Andrew Kofink <akofink@redhat.com>

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

**Review by vkrizan — APPROVED** [review]
Nice!

**Review by skateman — APPROVED** [review]
The Seal of Approval

File: test/tasks/status_test.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@ class RedisStatusTest < ActiveSupport::TestCase
 
 class KafkaStatusTest < ActiveSupport::TestCase
   test 'kafka:status fails without a kafka connection' do
-    TestProducer.stubs(:deliver_message).raises(Kafka::ConnectionError)
+    ApplicationProducer.stubs(:kafka).raises(Kafka::ConnectionError)
```
Comment (akofink) [inline test/tasks/status_test.rb:48]: Something about the way I was stubbing `Kafka::Client.any_instance` messed up Jenkins :thinking: I have no clue why, but this alternate stub seems to have fixed the issue.

Comment (Victoremepunto) [reply test/tasks/status_test.rb:48]: What the :duck:  dude ...

**Pull request merged**